src/Tribe/Utils/Taxonomy.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace Tribe\Utils;
+
+/**
+ * Class Taxonomy.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Utils
+ */
+class Taxonomy {
+	/**
+	 * Match any operand.
+	 *
+	 * @since TBD
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	const OPERAND_OR = 'OR';
+
+	/**
+	 * Match all operand.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	const OPERAND_AND = 'AND';
+
+	/**
+	 * Default operand for taxonomy filters.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	const DEFAULT_OPERAND = self::OPERAND_OR;
+
+	/**
+	 * Translates a given argument to repository arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @param string       $taxonomy Which taxonomy we are using to setup.
+	 * @param string|array $terms    Which terms we are going to use here.
+	 * @param string       $operand  Which is the operand we should use.
+	 *
+	 * @return array A fully qualified `tax_query` array, merge using `array_merge_recursively`.
+	 */
+	public static function translate_to_repo( $taxonomy, $terms, $operand = self::OPERAND_OR ) {
```

Comment:
I would change the method name to either `translate_to_orm_args` or `to_orm_args` or `to_repository_args`.